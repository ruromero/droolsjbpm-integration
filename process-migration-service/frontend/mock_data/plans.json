[
  {
    "id": 1,
    "name": "mockupTest1",
    "description": "mockup_",
    "mappings": {
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
        "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    },
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    }
  },
  {
    "id": 2,
    "name": "upgrade",
    "description": null,
    "mappings": {
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
        "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    },
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    }
  },
  {
    "id": 3,
    "name": "downgrade",
    "description": "mockup_",
    "mappings": {
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
        "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    },
    "source": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    },
    "target": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    }
  },
  {
    "id": 5,
    "name": "bbb",
    "description": "mockup_dddd",
    "mappings": {},
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "Mortgage_Process.MortgageApprovalProcess"
    },
    "target": {
      "containerId": "Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT",
      "processId": "Mortgage_Process.MortgageApprovalProcess"
    }
  },
  {
    "id": 8,
    "name": "aa",
    "description": "mockup_bb",
    "mappings": {},
    "source": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    }
  },
  {
    "id": 9,
    "name": "a",
    "description": "mockup_bfff",
    "mappings": {},
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT",
      "processId": "Mortgage_Process.MortgageApprovalProcess"
    }
  },
  {
    "id": 10,
    "name": "sdf",
    "description": "mockup_sdf",
    "mappings": {},
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT",
      "processId": "Mortgage_Process.MortgageApprovalProcess"
    }
  },
  {
    "id": 11,
    "name": "dfdfd",
    "description": "mockup_dfdf",
    "mappings": {},
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT",
      "processId": "Mortgage_Process.MortgageApprovalProcess"
    }
  },
  {
    "id": 12,
    "name": "upgrade",
    "description": "mockup_",
    "mappings": {
      "_D3E17247-1D94-47D8-93AD-D645E317B736":
        "_D3E17247-1D94-47D8-93AD-D645E317B736",
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
        "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    },
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    }
  },
  {
    "id": 13,
    "name": "upgrade",
    "description": "mockup_upgrade123",
    "mappings": {
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
        "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
    },
    "source": {
      "containerId": "evaluation_2.0.0-SNAPSHOT",
      "processId": "old_evaluation"
    },
    "target": {
      "containerId": "evaluation_3.0.0-SNAPSHOT",
      "processId": "new_evaluation"
    }
  }
]