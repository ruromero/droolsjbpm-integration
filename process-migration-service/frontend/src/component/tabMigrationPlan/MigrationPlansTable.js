import React from "react";
import PropTypes from "prop-types";
import {
  Table,
  Icon,
  OverlayTrigger,
  Tooltip,
  actionHeaderCellFormatter
} from "patternfly-react";

import WizardExecuteMigration from "./wizardExecuteMigration/WizardExecuteMigration";
import planClient from "../../clients/planClient";
import ExportPlanModal from "./ExportPlanModal";

export default class MigrationPlansTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showMigrationWizard: false,
      containerId: ""
    };
  }

  openMigrationWizard = id => {
    planClient.get(id).then(plan => {
      this.setState({
        showMigrationWizard: true,
        containerId: plan.source.containerId,
        planId: id
      });
    });
  };

  render() {
    const headerFormat = value => <Table.Heading>{value}</Table.Heading>;
    const cellFormat = (value, def) => {
      const parts = def.property.split(".");
      let newVal = def.rowData;
      parts.forEach(part => {
        newVal = newVal[part];
      });
      return <Table.Cell>{newVal}</Table.Cell>;
    };

    const tooltipExecute = (
      <Tooltip id="tooltip">
        <div>Execute migration</div>
      </Tooltip>
    );

    const tooltipDelete = (
      <Tooltip id="tooltip">
        <div>Delete</div>
      </Tooltip>
    );
    const tooltipEdit = (
      <Tooltip id="tooltip">
        <div>Edit</div>
      </Tooltip>
    );

    const planBootstrapColumns = [
      {
        header: {
          label: "ID",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "id"
      },
      {
        header: {
          label: "Plan Name",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "name"
      },
      {
        header: {
          label: "Description",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "description"
      },
      {
        header: {
          label: "Source container",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "source.containerId"
      },
      {
        header: {
          label: "Source process",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "source.processId"
      },
      {
        header: {
          label: "Target container",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "target.containerId"
      },
      {
        header: {
          label: "Target process",
          formatters: [headerFormat]
        },
        cell: {
          formatters: [cellFormat]
        },
        property: "target.processId"
      },
      {
        header: {
          label: "Actions",
          props: {
            rowSpan: 1,
            colSpan: 4
          },
          formatters: [actionHeaderCellFormatter]
        },
        cell: {
          formatters: [
            (value, { rowData }) => [
              <Table.Actions key="Actions_Migration">
                <OverlayTrigger overlay={tooltipExecute} placement={"bottom"}>
                  <Table.Button
                    bsStyle="link"
                    onClick={() => this.openMigrationWizard(rowData.id)}
                  >
                    <Icon type="fa" name="play" />
                  </Table.Button>
                </OverlayTrigger>
              </Table.Actions>,
              <Table.Actions key="Actions_Export">
                <ExportPlanModal plan={rowData} />
              </Table.Actions>,
              <Table.Actions key="Actions_Delete">
                <OverlayTrigger overlay={tooltipDelete} placement={"bottom"}>
                  <Table.Button
                    bsStyle="link"
                    onClick={() => this.props.onDeletePlan(rowData)}
                  >
                    <Icon type="fa" name="trash" />
                  </Table.Button>
                </OverlayTrigger>
              </Table.Actions>,
              <Table.Actions key="Actions_Edit">
                <OverlayTrigger overlay={tooltipEdit} placement={"bottom"}>
                  <Table.Button
                    bsStyle="link"
                    onClick={() => this.props.onEditPlan(rowData.id)}
                  >
                    <Icon type="fa" name="edit" />
                  </Table.Button>
                </OverlayTrigger>
              </Table.Actions>
            ]
          ]
        },
        property: "action"
      }
    ];

    return (
      <React.Fragment>
        <Table.PfProvider striped hover columns={planBootstrapColumns}>
          <Table.Header />
          <Table.Body rows={this.props.plans} rowKey="id" />
        </Table.PfProvider>
        {this.state.showMigrationWizard && (
          <WizardExecuteMigration
            isOpen={this.state.showMigrationWizard}
            onClose={() => this.setState({ showMigrationWizard: false })}
            containerId={this.state.containerId}
            kieServerId={this.props.kieServerId}
            planId={this.state.planId}
          />
        )}
      </React.Fragment>
    );
  }
}

MigrationPlansTable.propTypes = {
  plans: PropTypes.array.isRequired,
  kieServerId: PropTypes.string.isRequired,
  onEditPlan: PropTypes.func.isRequired,
  onDeletePlan: PropTypes.func.isRequired
};
