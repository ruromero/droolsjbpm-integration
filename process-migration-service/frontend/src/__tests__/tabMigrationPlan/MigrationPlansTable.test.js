import renderer from "react-test-renderer";
import React from "react";
import plans from "../../../mock_data/plans.json";
import MigrationPlansTable from "../../component/tabMigrationPlan/MigrationPlansTable";

test("MigrationPlansTable renders correctly using snapshot", () => {
  const myMock = jest.fn();
  const tree = renderer
    .create(
      <MigrationPlansTable
        plans={plans}
        kieServerId="kieServer1"
        onEditPlan={myMock}
        onDeletePlan={myMock}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
