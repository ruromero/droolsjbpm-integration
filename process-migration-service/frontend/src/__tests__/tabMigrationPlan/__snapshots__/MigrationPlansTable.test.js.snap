// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MigrationPlansTable renders correctly using snapshot 1`] = `
<table
  className="table table-striped table-hover"
>
  <thead>
    <tr>
      <th
        className=""
      >
        ID
      </th>
      <th
        className=""
      >
        Plan Name
      </th>
      <th
        className=""
      >
        Description
      </th>
      <th
        className=""
      >
        Source container
      </th>
      <th
        className=""
      >
        Source process
      </th>
      <th
        className=""
      >
        Target container
      </th>
      <th
        className=""
      >
        Target process
      </th>
      <th
        aria-label="Actions"
        className=""
        colSpan={4}
        rowSpan={1}
      >
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=""
      >
        1
      </td>
      <td
        className=""
      >
        mockupTest1
      </td>
      <td
        className=""
      >
        mockup_
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        2
      </td>
      <td
        className=""
      >
        upgrade
      </td>
      <td
        className=""
      />
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        3
      </td>
      <td
        className=""
      >
        downgrade
      </td>
      <td
        className=""
      >
        mockup_
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        5
      </td>
      <td
        className=""
      >
        bbb
      </td>
      <td
        className=""
      >
        mockup_dddd
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        8
      </td>
      <td
        className=""
      >
        aa
      </td>
      <td
        className=""
      >
        mockup_bb
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        9
      </td>
      <td
        className=""
      >
        a
      </td>
      <td
        className=""
      >
        mockup_bfff
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        10
      </td>
      <td
        className=""
      >
        sdf
      </td>
      <td
        className=""
      >
        mockup_sdf
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        11
      </td>
      <td
        className=""
      >
        dfdfd
      </td>
      <td
        className=""
      >
        mockup_dfdf
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess_1.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        Mortgage_Process.MortgageApprovalProcess
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        12
      </td>
      <td
        className=""
      >
        upgrade
      </td>
      <td
        className=""
      >
        mockup_
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className=""
      >
        13
      </td>
      <td
        className=""
      >
        upgrade
      </td>
      <td
        className=""
      >
        mockup_upgrade123
      </td>
      <td
        className=""
      >
        evaluation_2.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        old_evaluation
      </td>
      <td
        className=""
      >
        evaluation_3.0.0-SNAPSHOT
      </td>
      <td
        className=""
      >
        new_evaluation
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-play"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-export"
          />
        </button>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-trash"
            />
          </button>
        </div>
      </td>
      <td
        className="table-view-pf-actions"
      >
        <div
          className="table-view-pf-btn"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="fa fa-edit"
            />
          </button>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
